flytekit/tools/translator.py

```diff
@@ -617,6 +628,28 @@ def get_serializable_node(
 
 
 def get_serializable_array_node(
+    entity_mapping: OrderedDict,
+    settings: SerializationSettings,
+    node: FlyteLocalEntity,
+    options: Optional[Options] = None,
+) -> ArrayNodeModel:
+    array_node = node.flyte_entity
+    if array_node.min_successes is not None:
+        return ArrayNodeModel(
+            node=get_serializable_node(entity_mapping, settings, array_node, options=options),
+            parallelism=array_node.concurrency,
+            min_successes=array_node.min_successes,
+            execution_mode=array_node.execution_mode,
+        )
```

Comment:
why do we need to special case `min_successes`?

Reply:
Since we default min_success_ratio to 1.0, it's always set. When passing in both into the proto the one_of will take the last assigned value - which would be min_success_ratio if both are set.

Could also handle this by nulling out min_success_ratio if min_successes is set

Reply:
what do you think of the idea of changing the type of `min_success_ratio` in `map_task`? From `float` to `Optional[float]` and set its default value to `None` but at the same time check if both `min_successes` and `min_success_ratio` are set to `None` we "favor" `min_success_ratio` and set that to `1.0`?

This way we centralize this weird interaction between two different parameters in a single function and we don't really validate anything in `ArrayNodeModel` (so that instantiating objects of that type will always take the arguments as-is) and no more special-casing.